src/PngDecoder.ts

```diff
@@ -115,10 +155,51 @@ export default class PngDecoder extends IOBuffer {
       this.skip(4);
     }
   }
+  private decodeApngChunk(length: number, type: string): void {
+    const offset = this.offset;
+    if (type !== 'fdAT' && type !== 'IDAT' && this._writingDataChunks) {
+      this.pushDataToFrame();
+    }
+    switch (type) {
+      // 11.2 Critical chunks
+      case 'IHDR': // 11.2.2 IHDR Image header
+      case 'PLTE': // 11.2.3 PLTE Palette
+      case 'IDAT': // 11.2.4 IDAT Image data
+      case 'IEND': // 11.2.5 IEND Image trailer
+      case 'tRNS': // 11.3.2.1 tRNS Transparency
+      case 'iCCP': // 11.3.3.3 iCCP Embedded ICC profile
+      case textChunkName: // 11.3.4.3 tEXt Textual data
+      case 'pHYs': // 11.3.5.3 pHYs Physical pixel dimensions
+        this.decodeChunk(length, type);
+        this.offset = offset + length;
+        break;
+      case 'acTL':
+        this.decodeACTL();
+        break;
```

Comment:
can we not handle the specific `case`s for APNG here and in the `default`, call `this.decodeChunk(length, type)` to delegate without repeating the chunk names?